generic_linkedlist.c

```diff
@@ -0,0 +1,74 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+struct Node
+{
+    void *data;
+    struct Node *next;
+};
+
+struct Node *head = NULL;
+
+void insert(void *new_data, size_t data_size)
+{
+    struct Node *new_node = (struct Node *)malloc(sizeof(struct Node));
+    new_node->data = malloc(data_size);
+    new_node->next = (head);
+    memcpy(new_node->data, new_data, data_size);
+    head = new_node;
+}
+
+void printList(void (*fptr)(void *))
+{
+    struct Node *temp = head;
+    while (temp != NULL)
+    {
+        (*fptr)(temp->data);
+        temp = temp->next;
+    }
+}
+
+void printInt(void *n)
+{
+    printf(" %d", *(int *)n);
+}
+
+void printFloat(void *f)
+{
+    printf(" %f", *(float *)f);
+}
+
+int main()
+{
+    unsigned int_size = sizeof(int);
```

Comment:
can we divide the main function in further function like floatList, intList, charList. Then call these function via switch so user has a choice to choose the type of linkedlist